CMakeLists.txt

```diff
@@ -268,6 +277,12 @@ if(HARDENING)
     try_append_cxx_flags("-fcf-protection=full" TARGET hardening)
 
     if(MINGW)
+      # It happens to work in Autotools because Libtool passes -fstack-protector*
```

Comment:
I still don't think this should mention libtool, given it's not going to be used with CMake. Why not just state that this is required when using GCC < 11, and explain why it's not needed after, due to whatever changed in GCC 11. There's no need to mention libtool at all, because it works when using GCC 11, without libtool.

Reply:
We are still using GCC 10.5 in Guix. Wouldn't linking to libssp reintroduce https://github.com/bitcoin/bitcoin/issues/28104? Maybe stick to mirroring Libtool's behaviour and pass `-fstack-protector-all` to a linker context?

Reply:
I think this makes sense.

IIRC `-fstack-protector-all` tries to link in `libssp_nonshared.a` when it can. We could potentially use that instead of `-lssp`, but I believe it's a convention and not a guarantee.

I suppose we could also test for it in CMake: build 2 files from source with `-fstack-protector-all`, link them without it, and see if it fails. I think that's probably overkill though since the switch is (at least currently) harmless.

Agree with @fanquake here though. This is just working around a gcc bug, nothing to do with libtool. I'd prefer a comment like:
"Older versions of gcc required libssp to be linked, but this was fixed with gcc11. This can be removed when those older versions are no longer supported."

Reply:
Thanks! Implemented.